composer/core/state.py

```diff
@@ -134,12 +134,13 @@ def __init__(
             else:
                 train_dataloader = DataSpec(train_dataloader)
         self.train_data = train_dataloader
-        if not isinstance(eval_dataloader, DataSpec):
-            if isinstance(eval_dataloader, dict):
-                eval_dataloader = DataSpec(**eval_dataloader)
-            else:
-                eval_dataloader = DataSpec(eval_dataloader)
-        self.eval_data = eval_dataloader
+        for evaluator in evaluators:
+            if not isinstance(evaluator.dataset, DataSpec):
+                if isinstance(evaluator.dataset, dict):
+                    evaluator.dataset = DataSpec(**evaluator.dataset)
+                else:
+                    evaluator.dataset = DataSpec(evaluator.dataset)
+        self.evaluator_data = evaluators
```

Comment:
nit: 
```suggestion
        self._evaluators = evaluators
```
An evaluator is more than just the data :)